src/main/java/seedu/address/model/remark/Remark.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.remark;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Remark in MediConnect
+ * Guarantees: immutable; name is valid as declared in {@link #isValidRemark(String)}
+ */
+public class Remark {
+    public static final String MESSAGE_CONSTRAINTS = "";
+
+    public final String remark;
+
+    /**
+     * Constructs a {@code Remark}.
+     *
+     * @param remark A valid remark.
+     */
+    public Remark(String remark) {
+        requireNonNull(remark);
+        checkArgument(isValidRemark(remark), MESSAGE_CONSTRAINTS);
+        this.remark = remark;
+    }
+
+    /**
+     * Returns true if a given string is a valid NRIC.
+     */
+    public static boolean isValidRemark(String test) {
+        return true;
+    }
+
+    // TODO: implement validation check
+    @Override
+    public String toString() {
+        return remark;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof Remark)) {
+            return false;
+        }
+
+        Remark otherRemark = (Remark) other;
+        return remark.equals(otherRemark.remark);
+    }
+
+    @Override
+    public int hashCode() {
+        return remark.hashCode();
```

Comment:
you wanna use `Object.hash(remark)` here instead?

Reply:
i dont see any difference between each approach since Strings have its own implementation of hashCode already

Reply:
nope, was just thinking after adding a load of hashCodeMethod test today. but ya, if `hashcode()` and `Object.hash` produce the exact same hashCode then its fine

Reply:
They dont actually produce the same hashcode. `hashcode()` uses the String implementation which just generates it from a each character. `Objects.hash()` puts the items into an array and calls `Arrays.hashcode()` to generate the hashcode. But i believe either way does not violate lsp.

Reply:
as long as our hashCode test reflects the right method to get the String hashCode, then its fine